x-pack/plugins/fleet/server/routes/settings/index.ts

```diff
@@ -5,71 +5,68 @@
  * 2.0.
  */
 
-import type { TypeOf } from '@kbn/config-schema';
-
+import { parseExperimentalConfigValue } from '../../../common/experimental_features';
 import { API_VERSIONS } from '../../../common/constants';
 import type { FleetAuthzRouter } from '../../services/security';
-
 import { SETTINGS_API_ROUTES } from '../../constants';
-import type { FleetRequestHandler } from '../../types';
 import {
   PutSettingsRequestSchema,
   GetSettingsRequestSchema,
   GetEnrollmentSettingsRequestSchema,
+  GetSpaceSettingsRequestSchema,
+  PutSpaceSettingsRequestSchema,
 } from '../../types';
-import { defaultFleetErrorHandler } from '../../errors';
-import { settingsService, agentPolicyService, appContextService } from '../../services';
+import type { FleetConfigType } from '../../config';
 
 import { getEnrollmentSettingsHandler } from './enrollment_settings_handler';
 
-export const getSettingsHandler: FleetRequestHandler = async (context, request, response) => {
-  const soClient = (await context.fleet).internalSoClient;
-
-  try {
-    const settings = await settingsService.getSettings(soClient);
-    const body = {
-      item: settings,
-    };
-    return response.ok({ body });
-  } catch (error) {
-    if (error.isBoom && error.output.statusCode === 404) {
-      return response.notFound({
-        body: { message: `Settings not found` },
-      });
-    }
-
-    return defaultFleetErrorHandler({ error, response });
-  }
-};
-
-export const putSettingsHandler: FleetRequestHandler<
-  undefined,
-  undefined,
-  TypeOf<typeof PutSettingsRequestSchema.body>
-> = async (context, request, response) => {
-  const soClient = (await context.fleet).internalSoClient;
-  const esClient = (await context.core).elasticsearch.client.asInternalUser;
-  const user = appContextService.getSecurityCore().authc.getCurrentUser(request) || undefined;
+import {
+  getSettingsHandler,
+  getSpaceSettingsHandler,
+  putSettingsHandler,
+  putSpaceSettingsHandler,
+} from './settings_handler';
 
-  try {
-    const settings = await settingsService.saveSettings(soClient, request.body);
-    await agentPolicyService.bumpAllAgentPolicies(esClient, { user });
-    const body = {
-      item: settings,
-    };
-    return response.ok({ body });
-  } catch (error) {
-    if (error.isBoom && error.output.statusCode === 404) {
-      return response.notFound({
-        body: { message: `Settings not found` },
-      });
-    }
+export const registerRoutes = (router: FleetAuthzRouter, config: FleetConfigType) => {
+  const experimentalFeatures = parseExperimentalConfigValue(config.enableExperimental);
+  if (experimentalFeatures.useSpaceAwareness) {
+    router.versioned
+      .get({
```

Comment:
Should these new routes be secured with [access tags](https://docs.elastic.dev/kibana-dev-docs/key-concepts/designing-feature-privileges#control-access-to-api-endpoints-and-other-server-side-features)?

Does `fleetAuthz` provide adequate authorization?

Reply:
Yes `fleetAuthz` should provide adequate authorization, if you are interested in the details of it work it's implemented here https://github.com/nchaulet/kibana/blob/62923732f516feea8a048ec2ffe3195c885eade2/x-pack/plugins/fleet/server/services/security/fleet_router.ts#L52

Reply:
Excellent - thank you for the info - TIL!